src/h3/geo.pyx

```diff
@@ -35,84 +36,200 @@ cdef (double, double) coord2geo(h3lib.GeoCoord c):
     )
 
 
-cdef h3lib.Geofence make_geofence(geos):
+cpdef h3lib.H3int geo_to_h3(double lat, double lng, int res) except 1:
+    cdef:
+        h3lib.GeoCoord c
+
+    check_res(res)
+
+    c = geo2coord(lat, lng)
+
+    return h3lib.geoToH3(&c, res)
+
+
+cpdef (double, double) h3_to_geo(h3lib.H3int h) except *:
+    """Map an H3 cell into its centroid geo-coordinate (lat/lng)"""
+    cdef:
+        h3lib.GeoCoord c
+
+    check_addr(h)
+
+    h3lib.h3ToGeo(h, &c)
+
+    return coord2geo(c)
+
+
+cdef h3lib.Geofence make_geofence(geos, bool lnglat_order=False):
+    """
+
+    Parameters
+    ----------
+    geos : list or tuple
+        GeoFence: A sequence of >= 3 (lat, lng) pairs where the last
+        element may or may not be same as the first (to form a closed loop).
+        The order of the pairs may be either clockwise or counterclockwise.
+    lnglat_order : bool
+        If True, assume coordinate pairs like (lng, lat)
+        If False, assume coordinate pairs like (lat, lng)
+    """
     cdef:
         h3lib.Geofence gf
 
     gf.numVerts = len(geos)
 
-    # todo: figure out when/how to free this memory
     gf.verts = <h3lib.GeoCoord*> stdlib.calloc(gf.numVerts, sizeof(h3lib.GeoCoord))
 
-    for i, (lat, lng) in enumerate(geos):
+    if lnglat_order:
+        latlng = (g[::-1] for g in geos)
+    else:
+        latlng = geos
+
+    for i, (lat, lng) in enumerate(latlng):
         gf.verts[i] = geo2coord(lat, lng)
 
     return gf
 
 
+cdef free_geofence(h3lib.Geofence* gf):
+    if gf.verts:
+        stdlib.free(gf.verts)
```

Comment:
nit: the C `free` function on `NULL` is a no-op.

Reply:
good point. updated.